rainier-core/src/main/scala/com/stripe/rainier/optimizer/LBFGS.java

```diff
@@ -0,0 +1,755 @@
+/* Heavily modified from:
+ * RISO: an implementation of distributed belief networks. (Copyright 1999, Robert Dodier).
+ *
+ * License: Apache license, version 2.
+ */
+package com.stripe.rainier.optimizer;
+
+class LBFGS
```

Comment:
Per IRL, this should be private so no one can depend on the implementation and probably deserves a comment about being a pretty minimal port just to  get thread safety.

Reply:
Java's default is package-private, which is what we want.